include/circt/Dialect/SV/Expressions.td

```diff
@@ -22,3 +22,28 @@ def TextualValueOp : SVOp<"textual_value", [NoSideEffect]> {
      $string attr-dict `:` type($result)
   }];
 }
+
+// A list of things that the extract op knows with how to deal.
+def Extractable : Type<Or<[
+  CPred<"$_self.isa<::circt::sv::InterfaceType>()">
+]>, "Types from which ports or values can be extracted.">;
+
+def ExtractOp : SVOp<"extract", [NoSideEffect]> {
+  let summary = "SystemVerilog '.' operator";
+  let description = [{
+    The SystemVerilog '.' operator can be used to access
+    fields/members/modports in types (structs or unions) or interfaces. It is
+    not the '.' used in instantiations to connect signals (`.clk(clk)`).
+
+    Get the 'dataflow_in' modport on the 'handshake_example' interface:
+    ```
+    %ifaceModport = sv.extract @dataflow_in %iface : !sv.interface<@handshake_example>
```

Comment:
To me, it would make more sense if the syntax is `sv.extract $extractable $field`, to match the order of the arguments and how the type is printed (interface first, then modport). What do you think?

Reply:
Yeah, frankly I went with this since I couldn't get parsing the other way working. It was sucking up the `: <type>` as part of the symbol attribute.